src/components/Menu.jsx

```diff
@@ -12,6 +12,8 @@ const propTypes = {
   renderItemLabel: PropTypes.func,
   value: PropTypes.string,
   onSelect: PropTypes.func,
+  shouldTransferFocus: PropTypes.bool,
+  notifyBeforeFocusWrap: PropTypes.func,
 };
```

Comment:
For focus(itemIndex), wouldn't something like this work (combining what you had in setSelectedIndexToLast into focus):
```
  focus(itemIndex) {    
    if (typeof itemIndex === 'number')
      const {   
        options,  
      } = this.props;  
 
      this.selectedIndex = (itemIndex >= 0) ? itemIndex : options.length + itemIndex;
    }

    if (this.domNode) {  
      this.domNode.focus();  
    }  
 }
```